lib/jekyll/excerpt.rb

```diff
@@ -132,34 +132,47 @@ def render_with_liquid?
     # Returns excerpt String
 
     LIQUID_TAG_REGEX = %r!{%-?\s*(\w+)\s*.*?-?%}!m.freeze
-    MKDWN_LINK_REF_REGEX = %r!^ {0,3}\[[^\]]+\]:.+$!.freeze
+    MKDWN_LINK_REF_REGEX = %r!^ {0,3}(?:(\[[^\]]+\])(:.+))$!.freeze
 
     def extract_excerpt(doc_content)
       head, _, tail = doc_content.to_s.partition(doc.excerpt_separator)
-
-      # append appropriate closing tag(s) (for each Liquid block), to the `head` if the
-      # partitioning resulted in leaving the closing tag somewhere in the `tail` partition.
-      if head.include?("{%")
-        modified  = false
-        tag_names = head.scan(LIQUID_TAG_REGEX)
-        tag_names.flatten!
-        tag_names.reverse_each do |tag_name|
-          next unless liquid_block?(tag_name)
-          next if head =~ endtag_regex_stash(tag_name)
-
-          modified = true
-          head << "\n{% end#{tag_name} %}"
-        end
-        print_build_warning if modified
-      end
-
       return head if tail.empty?
 
-      head << "\n\n" << tail.scan(MKDWN_LINK_REF_REGEX).join("\n")
+      head = sanctify_liquid_tags(head) if head.include?("{%")
+      definitions = extract_mkdown_ref_link_defs(head, tail)
+      return head if definitions.empty?
+
+      head << "\n\n" << definitions.join("\n")
     end
 
     private
 
+    # append appropriate closing tag(s) (for each Liquid block), to the `head` if the
+    # partitioning resulted in leaving the closing tag somewhere in the `tail` partition.
+    def sanctify_liquid_tags(head)
+      modified  = false
+      tag_names = head.scan(LIQUID_TAG_REGEX)
+      tag_names.flatten!
+      tag_names.reverse_each do |tag_name|
+        next unless liquid_block?(tag_name)
+        next if head =~ endtag_regex_stash(tag_name)
+
+        modified = true
+        head << "\n{% end#{tag_name} %}"
+      end
+
+      print_build_warning if modified
+      head
+    end
+
+    def extract_mkdown_ref_link_defs(head, tail)
```

Comment:
```suggestion
    def extract_markdown_link_reference_defintions(head, tail)
```

If rubocop is going to complain here about the long function name, we should suppress that warning. I want longer, more readable method names more than I want to make rubocop happy.

Reply:
So this was not an issue in the end?